src/math/history.h

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2025 Team Dissolve and contributors
+
+#pragma once
+
+#include "base/serialiser.h"
+#include <memory>
+#include <vector>
+
+// Data History
+template <class T> class History
+{
+    private:
+    // Stored historical data
+    std::vector<std::unique_ptr<T>> history_;
+
+    public:
+    // Update history with supplied data and return current average
+    T average(const T &currentData, int averagingLength)
+    {
+        // Push the current data onto the history stack
+        history_.emplace_back(std::make_unique<T>(currentData));
+
+        // Prune old data to get to the averagingLength
+        while (history_.size() > averagingLength)
+            history_.erase(history_.begin());
+
+        // Perform averaging of the datasets that we have
+        T averaged;
+        auto weight = 1.0 / history_.size();
+        for (auto &data : history_)
+            averaged += *data * weight;
```

Comment:
```suggestion
        // How many items to average
        auto length = history.size() < averagingLength ? history.size() : averagingLength;

        // Perform averaging of the datasets that we have
        T averaged;
        auto weight = 1.0 / length;
        for (auto &data : std::span(history_.rbegin(), ristory_.rbegin()+length))
            averaged += *data * weight;
```

This is an alternate implementation that doesn't lose history.  This would allow us to look at multiple averaging lengths simultaneously (the current implementation essentially locks you into the shortest length).  The disadvantage is that it can grow without bound, though that could be fixed by replacing the vector with a ring buffer.

This isn't a necessary change - just a suggestion.

Reply:
Interesting thought. While I like the idea, my concern is that this could add a lot of bloat as many of the objects being stored here are pretty chunky (I'm looking at you, `PartialSet`).  As such, I would prefer to stick to the ultra-simple, shortest-length version for the time being.